src/runtime_src/core/tools/common/JSONConfigurable.cpp

```diff
@@ -6,37 +6,62 @@
 #include "tools/common/JSONConfigurable.h"
 #include "core/common/error.h"
 
-// Find children corresponding to target(s) and return them in a ptree.
-boost::property_tree::ptree
-JSONConfigurable::parse_configuration(
-  const std::vector<std::string>& targets,
+#include <iostream>
+#include <set>
+
+const std::map<std::string, std::string>
+JSONConfigurable::device_type_map = {
+  {"aie", "AIE"},
+  {"alveo", "Alveo"},
+  {"common", "common"}
+};
+
+// Parse configuration for children corresponding to current device categories before parsing for target(s).
+std::map<std::string, boost::property_tree::ptree>
+JSONConfigurable::parse_configuration_tree(
   const boost::property_tree::ptree& configuration)
 {
-  boost::property_tree::ptree output;
-  try {
-    for (const std::string& target: targets) {
-      for (const auto& configTree : configuration) {
-        for (const auto& config : configTree.second.get_child("contents")) {
-          const auto& pt = config.second;
-          if (boost::iequals(target, pt.get<std::string>("name")))
-            output.push_back({ configTree.second.get<std::string>("name"), pt });
+  std::map<std::string, boost::property_tree::ptree> target_mappings;
+  for (const auto& device_config_tree : configuration) { // Thanks ptree JSON parsing -_- We LOVE nested for loops!
```

Comment:
Let's use a comment ahead of line 25 that explains the deep nesting without thanking anyone :-)
Know that std::pair (or tuples) bind to `auto& [first, second]`  which can sometimes help readability when first and second are named appropriately.

Reply:
Good tip, I did this where I could!